tracer/test/Datadog.Trace.TestHelpers/MockHttpParser.cs

```diff
@@ -185,5 +191,88 @@ public async Task<bool> IsNewLine()
                 return false;
             }
         }
+
+        private class ChunkedEncodingReadContent(Stream stream)
+            : StreamContent(new ChunkedEncodingReadStream(stream), length: null);
+
+        private class ChunkedEncodingReadStream(Stream innerStream) : DelegatingStream(innerStream)
+        {
+            private readonly Stream _innerStream = innerStream;
+            private readonly StreamReaderHelper _helper = new(innerStream);
+            private readonly StringBuilder _sb = new();
+            private int _bytesRemainingInChunk = 0;
+            private State _state = State.AwaitingChunkHeader;
+
+            private enum State
+            {
+                AwaitingChunkHeader,
+                ReadingChunk,
+                ChunkFinished,
+                Complete,
+            }
+
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+                // YOLO
```

Comment:
😱

Reply:
It's only in tests 😉 Not worth duplicating all the HTTP parser etc IMO